packages/fluentui/react-builder/src/components/NarrationComputer.tsx

```diff
@@ -0,0 +1,262 @@
+* Should we also consider the "disabled" state?
+*/
+import { SRNC } from './SRNC-Definitions';
+import './SRNC-Rules-Win_JAWS';
+
+export class NarrationComputer {
+  computedParts: Record<string, string> = {
+    value: '',
+    name: '',
+    description: '',
+    type: '',
+    state: '',
+    position: '',
+    usage: '',
+  };
+
+  // Computes and returns the complete screen reader narration for the given element and platform.
+  async computeNarration(element: Element, platform: string): Promise<string> {
+    let definitionName = this.getDefinitionName(element, platform, 'stateRules');
+
+    // Retrieve the computed accessible node
+    const node = await (window as any).getComputedAccessibleNode(element);
+
+    // Compute and store all the narration parts
+    this.computeUsage(definitionName, element, platform);
+    this.computeDescription(definitionName, element, platform);
+    this.computeNameAndTitle(node, element);
+    this.computeValue(node);
+    this.computePosition(definitionName);
+    this.computeTypeAndState(node, definitionName, element, platform);
+
+    definitionName = this.getDefinitionName(element, platform, 'readingOrder');
+    const computedNarration = this.composeNarrationFromParts(definitionName, platform);
+    return computedNarration;
+  } // End computeNarration
+
+  // Returns the definition name based on the given element, platform and definition type.
+  getDefinitionName(element: Element, platform: string, definitionType: string): string {
+    // Determine the definitions source by the definition type
+    const definitionTypes: Record<string, any> = {
+      readingOrder: SRNC.readingOrder[platform],
+      stateRules: SRNC.stateRules[platform],
+    };
+    const definitions = definitionTypes[definitionType];
+
+    // Determine the definition name
+    let definitionName = '[default]';
+    const role = element.getAttribute('role');
+    let testName = `role=${role}`;
+    let definition = definitions[testName];
+    if (role && definition) {
+      // Begin if 1
+      // The definition name is determined by the "role" attribute and the definition exists
+      if (typeof definition === 'string') {
+        // Begin if 2
+        // The definition is a reference to another definition
+        definitionName = definition;
+      } else {
+        // else if 2
+        // The definition is a regular definition (not a reference to another definition)
+        definitionName = testName;
+      } // End if 2
```

Comment:
Can we make it a ternary for simplicity? I think it can be better to read.

Reply:
yes,yes, applied.